### PR TITLE
[runtime] add package to handle gomaxprocs and runtime settings

### PR DESCRIPTION
**Pull request by truthbk**
### What does this PR do?

This PR adds a `runtime` package. This package is geared towards allowing us to tweak the go runtime settings. 

For now it handles automatically setting `GOMAXPROCS` to the desired setting. If `GOMAXPROCS` is set as an environment variable, that value will take precedence. It relies on work done here: https://pkg.go.dev/go.uber.org/automaxprocs to set the value if no override is in place.

The heavy-lifting takes place in an `init()` routine so make a nameless import in your binary as follows to have the correct settings set:

```
import( 
        ...
        
	// runtime init routines
	_ "github.com/DataDog/datadog-agent/pkg/runtime"
)
```

### Motivation

Set the right number of system threads for the go runtime in heterogenous environments where cgroups could be in place (ie. kubernetes).

### Additional Notes

-

### Describe your test plan

TO-DO

### CODE REVIEW

**Review by ogaca-dd — APPROVED** [review]
LGTM!
I am just wondering why `GOMAXPROCS` supports millicpu (`m` prefix).

Note: changelog is missing.

File: releasenotes/notes/gomaxprocs-sane-settings-87c98222af7086d3.yaml [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+# section. This may mean repeating some details, but each section
+# must be readable independently of the other.
+#
+# Each section note must be formatted as reStructuredText.
+---
+enhancements:
+  - |
+    GOMAXPROCS is now set automatically to match the allocated CPU cgroup quota.
+    GOMAXPROCS can now also be manually specified and overridden in millicore units.
+    If no quota or GOMAXPROC value is set it will default to the original behavior.
```
Comment (remeh) [inline releasenotes/notes/gomaxprocs-sane-settings-87c98222af7086d3.yaml:13]: nit: typo

**Review by remeh — APPROVED** [review]
Nice addition! 📦

File: pkg/runtime/runtime.go [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+}
+
+// NumVCPU returns the number of virtualizes CPUs available to the process. It should be used instead of
+// runtime.NumCPU() in virtualized environments like K8s to ensure that processes don't attempt to
+// over-subscribe CPUs. For example, on a 16 vCPU machine in a docker container allocated 8 vCPUs,
+// runtime.NumCPU() will return 16 but NumVCPU() will return 8.
+func NumVCPU() int {
+	// Value < 1 returns the current value without altering it.
+	return runtime.GOMAXPROCS(0)
+}
```
Comment (olivielpeau) [inline pkg/runtime/runtime.go:79]: this function is unused, and doesn't return the result described in the docstring once `SetMaxProcs` has run. let's remove it?

File: pkg/runtime/runtime.go [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+	_ "go.uber.org/automaxprocs"
+)
+
+const (
+	gomaxprocsKey = "GOMAXPROCS"
+)
+
+func init() {
+	defer func() {
+		log.Infof("runtime: final GOMAXPROCS value is: %d", runtime.GOMAXPROCS(0))
```
Comment (olivielpeau) [inline pkg/runtime/runtime.go:23]: nit: probably better as a `debug` level message? But could be useful to include on status page (and/or in the flare)

**Review by olivielpeau — COMMENTED** [review]
Reviewing after the merge, I only had a couple of suggestions that can be addressed later in a follow up PR :)

**Pull request merged**